x/evm/keeper/keeper.go

```diff
@@ -36,8 +36,8 @@ import (
 	"github.com/evmos/ethermint/x/evm/types"
 )
 
-// CustomContractFn defines a custom precompiled contract generator with rules and returns a precompiled contract.
-type CustomContractFn func(params.Rules) vm.PrecompiledContract
+// CustomContractFn defines a custom precompiled contract generator with ctx, rules and returns a precompiled contract.
+type CustomContractFn func(sdk.Context, params.Rules) vm.PrecompiledContract
```

Comment:
what's the usage for ctx?

Reply:
to allow [adjust](https://github.com/crypto-org-chain/cronos/pull/1251/files#diff-43188752bb6a37285eb8110fb5cfce01db332fa5f2c285c6c70b785637e0eaf3R103) require gas with ctx params